continuous_integration/scripts/install.sh

```diff
@@ -23,6 +23,11 @@ if [[ ${UPSTREAM_DEV} ]]; then
         git+https://github.com/dask/distributed
 fi
 
+# Try nightly version of pyarrow
+conda uninstall --force pyarrow
+python -m pip install --extra-index-url https://pypi.fury.io/arrow-nightlies/ \
+    --pre pyarrow
```

Comment:
Note that you can also install `pyarrow` from `arrow-nightlies` conda channel. There all possible combinations that are on conda-forge are also available.